libs/minecraft/CommandUtils.arucas

```diff
@@ -0,0 +1,420 @@
+//Check Argument has Unique names, prevent client crash
+import FormatUtil from util.FormatUtils;
+import * from Minecraft;
+client = MinecraftClient.getClient();
+//functions for argParse
+fun checkArguments(iterable){
+    argNames = Set.of();
+    foreach( i : iterable){
+        if  (argNames.contains(i)){
+            throw new Error("Arguments name should have unique names!");
+        }
+        argNames.add(i);
+    }
+}
+
+//Very handy command builder
+//lambda as fun(){something;}
+//LinearCommand([ArgList], lambda).build() to get 'built' one, .debugBuild() to get 'string' format, .process() to process and register command.
+//new LinearCommand([['eDemiBolt'], ['x', 'Integer'],['y', 'Integer'],['z', 'Integer'], ['direction', 'Enum', Direction.type]], demiBoltTask).process();
+
+class FILOStack {
+    var asList;
```

Comment:
A First In Last Out is the opposite of a Stack which is First In First Out. What you want is a Queue which is already implemented in `util.Collections`. This should be removed.

Reply:
You changed it to a Deque, but I mean my point is that this shouldn't belong in this file. It belongs in `util.Collections`. Deques are both Stacks and Queues so should implement those interfaces too.

Reply:
it will now depend on Collections.Deque (pop and dequeue method)